projects/samples/devices/controllers/emitter_receiver/emitter_receiver.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 1996-2022 Cyberbotics Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+A controller which sends data from the emitter to the receiver while the robots avoid the obstacles.
+"""
+
+from controller import Robot
+
+
+class Controller(Robot):
+    SPEED = 6
+    COMMUNICATION_CHANNEL = 1
+    timeStep = 64
+
+    def __init__(self):
+        super(Controller, self).__init__()
+
+        # Get a handler to the motors and set target position to infinity (speed control).
+        self.left_motor = self.getDevice('left wheel motor')
+        self.right_motor = self.getDevice('right wheel motor')
+        self.left_motor.setPosition(float('inf'))
+        self.right_motor.setPosition(float('inf'))
+        self.left_motor.setVelocity(0.0)
+        self.right_motor.setVelocity(0.0)
+
+        # As we are using the same controller for the emitter and the receiver,
+        # we need to differentiate them
+        if self.getName() == 'MyBot emitter':
+            self.robot_type = 'EMITTER'
+            self.communication = self.getDevice('emitter')
+            # If the channel is not the right one, change it
+            channel = self.communication.getChannel()
+            if channel != self.COMMUNICATION_CHANNEL:
+                self.communication.setChannel(self.COMMUNICATION_CHANNEL)
+        elif self.getName() == 'MyBot receiver':
+            self.robot_type = 'RECEIVER'
+            self.communication = self.getDevice('receiver')
+            self.communication.enable(self.timeStep)
+        else:
+            print(f'Unrecognized robot name {self.getName()}. Exiting...')
+            return
+
+        self.ds0 = self.getDevice('ds0')
+        self.ds1 = self.getDevice('ds1')
+        self.ds0.enable(self.timeStep)
+        self.ds1.enable(self.timeStep)
+
+    def run(self):
+        message_printed = 0
+        while self.step(self.timeStep) != -1:
+            # The emitter simply sends the message but the receiver
+            # has to check if there is something before it can reads the buffer.
+            if self.robot_type == 'EMITTER':
+                self.communication.send("Hello!")
+            else:
+                # is there at least one packet in the receiver's queue?
+                if self.communication.getQueueLength() > 0:
+                    # read current packet's data
+                    buffer = self.communication.getData().decode()
```

Comment:
This line throws the following error:
```
AttributeError: 'str' object has no attribute 'decode'. Did you mean: 'encode'?
```

Moreover `Receiver.getData` seems to be deprecated and should not be used in an example